engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -1194,194 +1571,24 @@ static int SetAtlas(lua_State* L)
     CheckResource(L, g_ResourceModule.m_Factory, path_hash, "texturesetc");
 
     dmGameSystemDDF::TextureSet texture_set_ddf = {};
-    uint32_t frame_index_count                  = 0;
     uint32_t num_geometries                     = 0;
     uint32_t num_animations                     = 0;
-    float tex_width                             = 0;
-    float tex_height                            = 0;
 
     luaL_checktype(L, 2, LUA_TTABLE);
     lua_pushvalue(L, 2);
 
-    // Set texture resource path
-    lua_getfield(L, -1, "texture");
-    {
-        const char* texture_path         = luaL_checkstring(L, -1);
-        dmhash_t tex_canonical_path_hash = GetCanonicalPathHash(texture_path);
-        void* texture_res                = CheckResource(L, g_ResourceModule.m_Factory, tex_canonical_path_hash, "texturec");
-        assert(texture_res);
-        texture_set_ddf.m_Texture        = texture_path;
-        dmGraphics::HTexture texture     = (dmGraphics::HTexture) texture_res;
-        tex_width                        = dmGraphics::GetTextureWidth(texture);
-        tex_height                       = dmGraphics::GetTextureHeight(texture);
-    }
-    lua_pop(L, 1); // "texture"
+    dmGraphics::HTexture texture;
+    const char* texture_path;
+    CheckTextureResource(L, -1, "texture", &texture_path, &texture);
 
     // Note: We do a separate pass over the lua state to validate the data in the args table,
     //       this is because we need to allocate dynamic memory and can't use luaL_check** functions
     //       since they longjmp away so we can't release the memory..
-    ValidateSetAtlasArgumentsFromLua(L, &num_geometries, &num_animations);
-
-    texture_set_ddf.m_Geometries.m_Data  = new dmGameSystemDDF::SpriteGeometry[num_geometries];
-    texture_set_ddf.m_Geometries.m_Count = num_geometries;
-    memset(texture_set_ddf.m_Geometries.m_Data, 0, sizeof(dmGameSystemDDF::SpriteGeometry) * num_geometries);
-
-    texture_set_ddf.m_Animations.m_Data  = new dmGameSystemDDF::TextureSetAnimation[num_animations];
```

Comment:
Moved all of this into a helper function since we want to do the same validation when creating the atlas as here